master/buildbot/status/web/jsontestresults.py

```diff
@@ -0,0 +1,68 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+import json
+from twisted.python import log
+from buildbot.status.web.base import HtmlResource, path_to_builder, path_to_builders, path_to_codebases, path_to_build
+
+
+class JSONTestResource(HtmlResource):
+    def __init__(self, log, step_status):
+        HtmlResource.__init__(self)
+        self.log = log
+        self.step_status = step_status
+
+    def suite_command_line(self, suite):
+        result = 'perl ./Tools/UnifiedTestRunner/test.pl --suite=' + suite["name"]
```

Comment:
it will be nice not having the perl command here ? not sure why is needed it is best if this information is pass as a parameter